### ISSUE TITLE
Empty workspace causes GTD to throw errors

### ISSUE THREAD

**Issue by selene57** [nvim-neorg/neorg#353]
### Prerequisites

- [X] I have read [contribution guidelines](https://github.com/vhyrro/neorg/blob/main/docs/CONTRIBUTING.md)
- [X] I am using the latest version of the plugin
- [X] I am using either 0.6 stable or the latest compiled neovim version

### Neovim Version

NVIM v0.6.1

### Neorg setup

```require('neorg').setup {
            -- Tell Neorg what modules to load
            load = {
                ["core.defaults"] = {}, -- Load all the default modules
                ["core.gtd.base"] = {
                    config = {
                        -- workspace =   "example_gtd" , -- assign the workspace,
                        workspace = "home",
                        exclude = { "notes/" }, -- Optional: all excluded files from the workspace are not part of the gtd workflow
                        projects = {
                          show_completed_projects = false,
                          show_projects_without_tasks = false,
                        },
                        custom_tag_completion = true,
                    },
                },
                ["core.norg.completion"] = { config = { engine = "nvim-cmp" } },
                ["core.norg.concealer"] = {}, -- Allows for use of icons
                ["core.norg.dirman"] = { -- Manage your directories with Neorg
                    config = {
                        workspaces = {
                            home = "~/neorg",
                            notes = "~/neorg/notes"
                        },
                        index = "index.norg"
                    }
                }
            },
          }
```

### Actual behavior

Initiating :NeorgStart throws the following error:

```Error executing vim.schedule lua callback: .../neorg/lua/neorg/modules/core/gtd/queries/retrievers.
lua:94: bad argument #1 to 'pairs' (table expected, got nil)
stack traceback:
        [C]: in function 'pairs'
        .../neorg/lua/neorg/modules/core/gtd/queries/retrievers.lua:94: in function 'get'
        ...r/start/neorg/lua/neorg/modules/core/gtd/base/module.lua:166: in function <...r/start/ne
org/lua/neorg/modules/core/gtd/base/module.lua:160>
Press ENTER or type command to continue
[neorg] (16:01:37)
...tart/neorg/lua/neorg/modules/core/gtd/helpers/module.lua:35
Press ENTER or type command to continue
No files found in home workspace.
Press ENTER or type command to continue
```

Other details:

This is on startup, meaning the workspace the GTD module has selected as it's workspace, is empty.

### Expected behavior

Initiating :NeorgStart when GTD points to an empty directory/workspace should create the inbox.neorg and index.neorg files in that directory. Or failing that, give you a clear English prompt that those files should be created.

### Steps to reproduce

1. Set the GTD module to a workspace that has no files in it.
2. Initiate nvim
3. Execute :NeorgStart

### Other information

This error is not thrown, when you copy/paste over the index.norg / inbox.norg files from the example config.

There are other errors associated with having an empty GTD workspace directory as well.

### Help

No

**Comment by danymat** [comment]
Hello ! 
Thanks for your feedback, I updated the error log.
By the way, I'm still not sure whether we should auto-create the `index.norg` file and `inbox.norg` file, what do you think ?

**Comment by selene57** [comment]
Personally, I think auto creating a template is what I'd pick, if it was up to me. The vast majority of users are already going to do that. It could prompt you with something like:

"No files found in your workspace for the gtd module. Would you like to create the default template files?"

Something like that I think would be a good compromise I think.

**Comment by danymat** [comment]
It's a good idea, in fact. We could use the popups to prompt user. I will think of that

**Comment by danymat** [comment]
Hello @selene57, 

Neorg now automatically creates required files (inbox or index) if they are missing from gtd workspace. 
This is implemented in [1f0e2e8](https://github.com/nvim-neorg/neorg/commit/1f0e2e84fa256cfcfb9af64797beec8a74cc19f1)

I'm closing this issue as it was the last thing to do, feel free to reopen if you find any bug in the implementation
